src/jvmMain/kotlin/viewmodel/FilesViewModel.kt

```diff
@@ -4,52 +4,64 @@ import androidx.compose.runtime.derivedStateOf
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.setValue
+import data.CodeAttributeRecord
+import data.LoadUtil
 import data.StateTracker
-import proguard.classfile.ClassPool
-import proguard.classfile.attribute.Attribute
-import proguard.classfile.attribute.visitor.AllAttributeVisitor
-import proguard.classfile.attribute.visitor.AttributeNameFilter
-import proguard.classfile.visitor.AllClassVisitor
-import proguard.classfile.visitor.AllMethodVisitor
-import proguard.classfile.visitor.ClassPoolFiller
-import proguard.evaluation.PartialEvaluator
-import proguard.evaluation.util.jsonPrinter.JsonPrinter
-import proguard.io.ClassFilter
-import proguard.io.ClassReader
-import proguard.io.DataEntrySource
-import proguard.io.FileSource
-import proguard.io.JarReader
 import java.nio.file.Path
 import kotlin.io.path.extension
 
 /**
  * A model to keep track and manage the files that are opened in the visualizer.
+ * When the currentCodeAttribute has not been loaded, this will load it for you
  */
 class FilesViewModel {
-    var files by mutableStateOf(emptyList<File>())
+    var files by mutableStateOf(emptyMap<Path, Map<String, Map<String, CodeAttributeViewModel?>>>())
         private set
 
-    private fun addFile(file: File) {
-        files = files.plus(file)
-    }
+    var curPath by mutableStateOf<Path?>(null)
 
-    fun closeFile(index: Int) {
-        files.getOrNull(index)?.let {
-            files = files.minus(it)
-        }
-    }
+    var curClazz by mutableStateOf<String?>(null)
 
-    private var fileIndex by mutableStateOf(0)
+    var curMethod by mutableStateOf<String?>(null)
 
-    private val currentFile by derivedStateOf { files.getOrNull(fileIndex) }
+    private fun addFile(file: Path, codeAttributes: List<CodeAttributeRecord>) {
+        val some = codeAttributes.groupBy { it.clazz }
+            .mapValues {
+                it.value.groupBy { map -> map.method }.mapValues { map -> CodeAttributeViewModel(map.value[0]) }
+            }
 
-    private var codeAttributeIndex by mutableStateOf(0)
+        files = files.plus(Pair(file, some))
+    }
 
-    val currentCodeAttribute by derivedStateOf { currentFile?.codeAttributeViewModels?.getOrNull(codeAttributeIndex) }
+    fun closeFile(path: Path) {
+        files = files.minus(path)
+        if (curPath == path) {
+            curPath = null
+            curClazz = null
+            curMethod = null
+        }
+    }
 
-    fun selectCodeAttribute(fileIndex: Int, attributeIndex: Int) {
-        this.fileIndex = fileIndex
-        this.codeAttributeIndex = attributeIndex
+    val currentCodeAttributeViewModel by derivedStateOf {
+        curPath?.let { path ->
+            curClazz?.let { clazz ->
+                curMethod?.let { method ->
```

Comment:
how about that if statement? :upside_down_face:

Reply:
![image](https://github.com/Mouwrice/proguard-core-visualizer/assets/35114273/ba3e2c6d-c6d1-43d1-87cb-b723468f528c)